src/snarl_seed_clusterer.hpp

```diff
@@ -444,15 +525,14 @@ class NewSnarlSeedClusterer {
         //SeedIndex is used to store the seeds
         //left offset is the distance from the left side of the structure
         template <typename SeedIndex>
-        void cluster_seeds_on_linear_structure(TreeState& tree_state, NodeClusters& node_clusters, vector<SeedIndex>& seed_indices, 
+        void cluster_seeds_on_linear_structure(TreeState& tree_state, NodeClusters& node_clusters,
+                const typename vector<SeedIndex>::iterator& range_start,
+                const typename vector<SeedIndex>::iterator& range_end,
                 size_t structure_length, std::function<std::tuple<size_t, size_t, size_t>(const SeedIndex&)>& get_offset_from_seed_index, bool skip_distances_to_ends) const;
 
         //Compare two children of the parent and combine their clusters, to create clusters in the parent
-        //This assumes that the first node hasn't been seen before but the second one has, so all of the
-        //first node's clusters get added to the parent but assume that all of the second ones are already
-        //part of the parent
-        //old_distances contains the distances for cluster heads in the children, 
-        //since the distances in tree_state.read_cluster_heads_to_distances will get updated
+        //child_distances contains the distances for cluster heads in the children, 
+        //since the distances in the seeds will get updated to be the distances in the parent
         //First child is true if this is the first time we see child_clusters1. If first_child is true and this is 
         //a snarl, then we need to update the snarl's distances to its parents
         void compare_and_combine_cluster_on_child_structures(TreeState& tree_state, NodeClusters& child_clusters1,
```

Comment:
If the node clustering records knew if they were roots or whether they had their distances to parents filled in yet, you could drop the flag arguments.